src/inc/corhlpr.h

```diff
@@ -40,10 +40,15 @@ inline void DECLSPEC_NORETURN THROW_OUT_OF_MEMORY()
 #else
 #define NEW_NOTHROW(_bytes) new BYTE[_bytes]
 #define NEW_THROWS(_bytes) __CorHlprNewThrows(_bytes)
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winvalid-noreturn"
```

Comment:
How did you hit this warning? When we build coreclr repo on Unix, it is warning free (warnings would cause build failure).

Reply:
@janvorli I have no idea why it works on your side: https://godbolt.org/z/fRFRH7

Reply:
Ah, this code is not compiled when the header is used in CoreCLR. It gets compiled only for external users (when  _BLD_CLR is not defined). 
Can you please change the fix to add `_ASSERTE(!"unreached");` after the RaiseException instead of disabling the warning?

Reply:
@janvorli _ASSERTE() doesn't help to solve this issue. In Release it is defined to `((void)0)`, and in Debug it's DebugBreak() which isn't [[noreturn]].

Reply:
Ah, right. We also have _UNREACHABLE() macro defined somewhere. Can you please try if it is visible here?

Reply:
I found few places where UNREACHABLE is defined:
* src/ToolBox/SOS/Strike/util.h (not visible),
* src/inc/check.h (seems we cannot use it here because it includes src/inc/clrtypes.h which conflicts with pal_mstypes.h),
* and src/inc/gcinfodecoder.h (looks like it's defined for its own purposes).

May I just use __builtin_unreachable() if !_MSC_VER, or create a new not platform-dependent file with that?

Reply:
@k15tfu I think that copying the __UNREACHABLE definition from check.h into this header would be ok.

Reply:
Sigh, that collides with other definition of __UNREACHABLE from `src/ToolBox/SOS/Strike/util.h`. We actually have two (and slightly different )definitions of unreachable in our codebase. So it seems that the right cure is to unify it (for Unix use the version from check.h), put it into a new header located at src/inc and include it from all three places.

Reply:
@janvorli Done. Only two definitions left (in check.h & gcinfodecoder.h).